syncer/main.py

```diff
@@ -54,137 +127,369 @@ def __init__(self, webinterface, connector):
         self._config = self.read_config(Syncer.default_config)
         self._config_checker = PluginConfigChecker(Syncer.config_description)
 
-        self._token = None
+        self._polling_interval = 60
+        self._gateways = {}
+        self._name = ''
+        self._output_mapping = {}
+        self._shutter_mapping = {}
         self._enabled = False
-        self._previous_outputs = set()
-        self._read_config()
+        thread = Thread(target=self._read_config)
+        thread.start()
+
+        self.connector.output.subscribe_status_event(handler=self.handle_output_status, version=2)
+        self.connector.shutter.subscribe_status_event(handler=self.handle_shutter_status, version=2)
 
         logger.info("Started Syncer plugin")
 
     def _read_config(self):
-        self._ip = self._config.get('gateway_ip', '')
-        self._username = self._config.get('username', '')
-        self._password = self._config.get('password', '')
+        self._enabled = False
+        self._polling_interval = self._config.get('polling_interval', 60)
+        self._name = self._config.get('local_name', '')
+        self._gateways = {}
+        output_mapping = {}
+        shutter_mapping = {}
+        _enabled = False
+        for gateway in self._config.get('gateways', ''):
+
+            ip = gateway.get('gateway_ip', '')
+            username = gateway.get('username', '')
+            password = gateway.get('password', '')
+
+            if gateway.get('remote_name', '') == '':
+                gw_name = ip
+            else:
+                gw_name = gateway.get('remote_name')
+
+            headers = {
+                'X-Requested-With': 'OpenMotics plugin: Syncer'
+            }
+            endpoint = 'https://{0}'.format(ip)
 
-        self._sensor_mapping = {}
-        for entry in self._config.get('sensors', []):
+            enabled = ip != '' and username != '' and password != ''
+            # If one GW is enabled, then the plugin will be enabled as well. But this is done after gathering all the
+            # config of the other GWs.
+            if enabled:
+                _enabled = True
```

Comment:
is there a reason why you have 3 enabled flags? ( 'enabled', '_enabled', 'self._enabled' )
This is very confusing to read by someone else. As far as I can see, there is no need for 3 separate flags?

Reply:
If you know a cleaner way to do this, suggestions are always welcome @ptuytsch

Reply:
let's leave it for now. When reading over it with a fresh mind, it is doable :)